src/OrleansSQLUtils/Storage/DbConnectionFactory.cs

```diff
@@ -1,39 +1,62 @@
 using System;
 using System.Collections.Concurrent;
-using System.Data;
+using System.Collections.Generic;
 using System.Data.Common;
+using System.Linq;
+using System.Reflection;
 
 namespace Orleans.SqlUtils
 {
-    /// It turns out that <see cref="DbProviderFactories.GetFactory(string)"/> uses reflection to fetch the singleton instance on every call.
-    /// this class caches the references to all loaded factories
+    /// This class caches the references to all loaded factories
     internal static class DbConnectionFactory
     {
         private static readonly ConcurrentDictionary<string, CachedFactory> factoryCache =
             new ConcurrentDictionary<string, CachedFactory>();
 
+        private static readonly Dictionary<string, Tuple<string, string>> providerFactoryTypeMap =
+            new Dictionary<string, Tuple<string, string>>
+            {
+                { AdoNetInvariants.InvariantNameSqlServer, new Tuple<string, string>("System.Data.SqlClient", "System.Data.SqlClient.SqlClientFactory") },
+                { AdoNetInvariants.InvariantNameMySql, new Tuple<string, string>("MySql.Data", "MySql.Data.MySqlClient.MySqlClientFactory") },
+                { AdoNetInvariants.InvariantNameOracleDatabase, new Tuple<string, string>("System.Data.SqlClient", "System.Data.SqlClient.SqlClientFactory") },
+                { AdoNetInvariants.InvariantNamePostgreSql, new Tuple<string, string>("Npgsql", "Npgsql.NpgsqlFactory ") },
+                { AdoNetInvariants.InvariantNameSqlLite, new Tuple<string, string>("Microsoft.Data.SQLite", "Microsoft.Data.SQLite.SqliteFactory") },
+            };
+
         private static CachedFactory GetFactory(string invariantName)
         {
             if (String.IsNullOrWhiteSpace(invariantName))
             {
                 throw new ArgumentNullException(nameof(invariantName));
             }
 
-            // Seeks for database provider factory classes from GAC or as indicated by
-            // the configuration file, see at <see href="https://msdn.microsoft.com/en-us/library/dd0w4a2z%28v=vs.110%29.aspx">Obtaining a DbProviderFactory</see>.       
+            if (!providerFactoryTypeMap.TryGetValue(invariantName, out Tuple<string, string> providerFactoryDefinition))
+                throw new InvalidOperationException($"Database provider factory with '{invariantName}' invariant name not supported.");
```

Comment:
not compiling on my machine. maybe I'm crazy, but I did build netstandard after git clean -xdf